srcjs/shinyapp.js

```diff
@@ -4,6 +4,9 @@ var ShinyApp = function() {
   // Cached input values
   this.$inputValues = {};
 
+  // Input values at initialization (and reconnect)
+  this.$initialInput = {};
```

Comment:
`this.$initialInput` and `this$inputValues` should not keep `opts`, because it would prevent the `el` from being garbage collected. The `opts` are only used in the input decorator stack; there's no need to keep them outside of that.